### ISSUE TITLE
Nether portal fails to create its overworld pair

### ISSUE THREAD

**Issue by khankul** [SwitchCraftCC/issues#167]
### Describe the bug

A nether portal will not be linked to the overworld if it's built too far away in the nether. I have not measured the coordinate threshold for it.

### To reproduce

1. Go to e.g. [8000, 8000] in the nether
2. Build and activate you nether portal
3. Step into it
4. Fail to properly teleport to the overworld
5. Notice that the corresponding nether portal was not created

### Expected behavior

The portal should get created in the overworld and paired with the nether one.

### Please confirm

- [X] This is not an exploit

**Comment by Wojbie** [comment]
You do realize that 8000,8000 in nether would be 64000,64000 in overworld and that is outside of playable area?
Does it still happen with nether portal that targets inside the world border?

**Comment by khankul** [comment]
As far as I'm aware, the vanilla behavior is to spawn the corresponding portal near the overworld border. I don't know if it's documented anywhere, but I remember it happening pretty constantly. Here's the result of me creating a portal at nether coordinates [-2035, 86] and passing through it. On SwitchCraft.
![image](https://github.com/SwitchCraftCC/issues/assets/93801865/7c987cff-5a79-48eb-8603-b48f87e20310)
The supposed overworld portal coordinates also go beyond the world size (by a few thousand blocks no less), but the portal still generates correctly.

**Comment by scmcgowen** [comment]
Yes, that is the vanilla behavior, as it has worked as your comment has stated on other servers where I’ve been playing on and had to leave the nether
However the reverse link is not created, causing a one way portal from hell back to the Overworld but it won’t bring you back to that position in hell

**Comment by khankul** [comment]
Another note: when teleporting using a link-failed portal, after the "Loading world" screen (which takes much longer than usual), the client is "tricked" into thinking that the player was teleported to the overworld, and will be waiting for chunks that will never come. Interestingly enough, while in this limbo state, the player is visibly absent from both worlds.
